packages/forms/src/Components/BelongsToSelect.php

```diff
@@ -126,7 +128,9 @@ public function relationship(string | Closure $relationshipName, string | Closur
             if ($component->hasOptionLabelFromRecordUsingCallback()) {
                 return $relationshipQuery
                     ->get()
-                    ->map(fn (Model $record) => $component->getOptionLabelFromRecord($record))
+                    ->mapWithKeys(fn (Model $record) => [
+                        $record->{$relationship->getOwnerKeyName()} => $component->getOptionLabelFromRecord($record)
```

Comment:
Do we need `$relationship->getOwnerKeyName()` here? Isn't that the same as `$record->getKey()`?
Before we had:

```suggestion
                        $record->getKey() => $component->getOptionLabelFromRecord($record)
```